### ISSUE TITLE
Unify mecanics

### ISSUE THREAD

**Issue by BinaryBrain** [BinaryBrain/Elodie-Dream#25]
At this point, every physics rules of the game depends on frame per seconds of the game. This should be avoided to be sure that everybody can finish it with the same level of difficulty.

**Comment by jrabasco** [comment]
It is a more complicated problem I guess. The physics should scale (all the speeds are weighed depending on the time in seconds between two frames) in function of the FPS rate, which appears not to exactly be the case. For instance nearly everybody has the same speed but not everybody jumps the same way. I would first take a look at the way the gravity depends on the fps (if the constants are correctly multiplied by the time it is not a poblem, nut maybe we should take time^2 into account for this one ? or square root of time). I think this should fix most of the problem.

**Comment by Protectator** [comment]
This is, for me, the most important issue to solve right now.
I think there are many solutions (I'll just tell general ideas, not going into implementation at all)
1. best overall but wayyy more complex solution : Differentiate a tick and a frame. A tick is an unity of time in the game (for example 60 ticks/second) in terms on physics. A frame is an image displayed at screen. The tick rate would be constant so that the mechanics in the game always run at the same speed and we could specify them without having to deal with computer performance. Frames are computed and displayed as fast as they can in another thread. The system would always compute ticks, and display frames when it can. So that when there isn't sufficient resources, the game would diminish framerate while always running at the same speed. But I think this solution is too mush effort-demanding for the final result.
2. Correct the actual solution. I don't know exactly what's wrong in the current implementation, but if it is managed to be corrected and that every computer has the same physics, I'm ok with it :)
3. Make a unit of time that represents always the same unit of time in terms of mechanics in-game, but not always the same in terms of the user. For example, a frame would be always computed as 1/60 of a second, and displayed as either 1/60 of a second if it is possible, or asap else. This would result of the whole game slowing down in case of performance issues.

tl;dr : Choose one : (1. Drop FPS when resources are unsufficient / 2. Correct current implementation / 3. Slow speed when resources are unsifficient)

**Comment by BinaryBrain** [comment]
1) This is way to complicated to manage (concurrence problem, dynamically
balancing the number of tick and fps, synchronizing both).
3) This is a bad idea for many reason: if the game slows down from 60 to 30
fps, it is still fluid but the time is divided by two, making it terribly
annoying (even from 50 to 40 it's terrible). Plus, running the game on a
toaster is easier, making the game unbalanced between players.

**Comment by Protectator** [comment]
did not mention it, but I also really think the 2nd option is the way to go. I was just listing options I thought about

**Comment by Protectator** [comment]
Commit https://github.com/BinaryBrain/Elodie-Dream/commit/7715afce3a083c45c3af746111f286d02f22cb6e is using 3rd solution for now.

**Comment by Protectator** [comment]
Btw, we are not alone, this thread contains some useful informations : http://stackoverflow.com/questions/17411/how-do-you-separate-game-logic-from-display
And [this article](http://www.koonsolo.com/news/dewitters-gameloop/) pretty much explains each possibility we discussed here (with some variations to each)
